src/storage-controller/src/command_wals.rs

```diff
@@ -157,6 +157,9 @@ where
                 .await;
         }
 
+        // Finalize any other shards that have been added to the WAL.
+        self.finalize_shards().await;
```

Comment:
Looks like the other call to `finalize_shards` is protected by the `enable_storage_shard_finalization` config, so we should probably do the same here (potentially by moving the config check into the impl of `fn finalize_shards`?)

Reply:
Makes sense! Moved the check into the method so it can't be forgotten...